remove Katello CA certs in clean_katello_agent

sideangleside opened the pull request:
In https://github.com/Katello/katello-client-bootstrap/pull/120, we added support for gracefully failing in the event that a system was already registered. (Done via checking for the presence of `/etc/rhsm/ca/katello-server-ca.pem`) As newer versions of the katello-ca-consumer RPM no longer deploy `/etc/rhsm/ca/katello-server-ca.pem` directly, but call `katello-rhsm-consumer`, which deploys the file, the **clean_katello_agent** function needs to delete *both* the `/etc/rhsm/ca/katello-server-ca.pem` file AND the RPM.

sideangleside committed:
remove Katello CA certs in clean_katello_agent

evgeni reviewed (changes requested):
https://docs.python.org/2/library/os.html#os.unlink

sideangleside committed:
Add delete_file() and delete_directory() functions
update clean_katello_agent(), clean_puppet() and clean_environment() functions to use new deletion functions

sideangleside committed:
minor pep8 fixups
